tests/integration/README.md

```diff
@@ -0,0 +1,21 @@
+#### Integration tests
+
+The continuous integration runs a set of [gherkin integration tests](https://github.com/open-feature/test-harness/blob/main/features).
+If you'd like to run them locally, first pull the `test-harness` git submodule
+```
+git submodule update --init --recursive
+```
+then build your flagd image
+```
+make docker-build
+```
+then run your built flagd image
+```
+docker run -p 8013:8013 -v $PWD/test-harness/testing-flags.json:/testing-flags.json ghcr.io/open-feature/flagd:latest start -f file:/testing-flags.json
+```
+and finally run
+```
+make integration-test
+```
+
+Note: Testing against the flagd binary directly (rather than the docker image) introduces test flakiness.
```

Comment:
I'm going to investigate why this is happening

Reply:
I've tried increasing the sleep between the file change and call to the previously cached flag to no avail, it still passes/fails intermittently. This leads me to believe the behaviour of my mac's file change notifications system is inconsistent with the behaviour of our distroless image's system.
I'm currently using macOS Monterey, I'm going to upgrade to Ventura tonight and revisit to see if the issue persists.
Either way, I will note that the expected behaviour is only stably supported when using the built image directly.

Reply:
There's oddities in windows too. A colleague was unable to get events working at all if they mounted a directory in docker in windows. They also were unable to get it to work if they mounted a shared WSL/Windows file in docker. Only a non-shared, WSL, file worked.

They had no issue in WSL without docker though, which would be using emulated inotify events, I guess.

You might want to look at the doc for https://github.com/fsnotify/fsnotify... maybe it can be improved, but we should probably say fsevents in anything other than linux are "best effort".

Reply:
Created an issue to investigate further: https://github.com/open-feature/flagd/issues/313